File: source/internal/server/api-types/video-data.go
package apitypes

import (
	"ova-cli/source/internal/datatypes"
	"time"
)

// Define a new struct for video status
type UserVideoStatus struct {
	IsWatched bool `json:"isWatched"`
	IsSaved   bool `json:"isSaved"`
}
type VideoStats struct {
	Views     int `json:"views"`
	Downloads int `json:"downloads"`
}

type VideoDataAPIResponse struct {
	VideoID              string                `json:"videoId"`
	FileName             string                `json:"fileName"`
	Tags                 []string              `json:"tags"`
	Markers              []MarkerDataRequest   `json:"markers"`
	Codecs               datatypes.VideoCodecs `json:"codecs"`
	IsCooked             bool                  `json:"isCooked"`
	OwnerAccountUsername string                `json:"ownerAccountUsername"`
	VideoStats           VideoStats            `json:"stats"`
	IsPublic             bool                  `json:"isPublic"`
	UploadedAt           time.Time             `json:"uploadedAt"`
	VideoStatus          UserVideoStatus       `json:"userVideoStatus"`
}

File: web/src/ova-angular-sdk/core-types/video-data.ts
import { VideoMarker } from './video-marker';

export interface VideoCodecs {
  format: string;
  durationSec: number;
  frameRate: number;
  isFragment: boolean;
  resolution: Resolution;
  videoCodec: string;
  audioCodec: string;
}

export interface Resolution {
  width: number;
  height: number;
}

export interface userVideoStatus {
  isWatched: boolean;
  isSaved: boolean;
}

export interface VideoStats {
  views: string;
  downloads: string;
}

export interface VideoData {
  videoId: string;
  title: string;
  tags: string[];
  uploadedAt: string;
  stats: VideoStats;
  codecs: VideoCodecs;
  isCooked: boolean;
  uploaderId: string;
  isPublic: boolean;
}
